src/js/postHandler.js

```diff
@@ -145,5 +145,5 @@ async function makeRequest(targetURL, reqData, proxyURL) {
   if (response.status < 200 || response.status >= 300) {
     throw new Error(data);
   }
-  return data;
+  return JSON.parse(data);
```

Comment:
What happens if the POST requests returns a plain text value, instead of a JSON value, does the plain response will be printed?

Reply:
Ah, no actually. `JSON.parse` will fail. 
It should be wrapped with try catch bloc. If parsing fails, return the data itself.